primitives/state-machine/src/overlayed_changes.rs

```diff
@@ -60,6 +69,92 @@ pub struct OverlayedChangeSet {
 	pub children: HashMap<Vec<u8>, BTreeMap<Vec<u8>, OverlayedValue>>,
 }
 
+/// A storage changes structure that can be generated by the data collected in [`OverlayedChanges`].
+///
+/// This contains all the changes to the storage and transactions to apply theses changes to the
+/// backend.
+pub struct StorageChanges<Transaction, H: Hasher, N: BlockNumber> {
+	/// All changes to the main storage.
+	///
+	/// A value of `None` means that it was deleted.
+	pub main_storage_changes: Vec<(Vec<u8>, Option<Vec<u8>>)>,
+	/// All changes to the child storages.
+	pub child_storage_changes: Vec<(Vec<u8>, Vec<(Vec<u8>, Option<Vec<u8>>)>)>,
```

Comment:
Can we get a typedef for this? (or actually multiple typedefs), it means close to nothing currently.

Reply:
I'm with you. We can create an issue for this. I really don't want to touch this as well in this pr. As this probably will require to change again a lot of lines :( (I did not come up with these opaque types)

Reply:
Fine with me. Just created: https://github.com/paritytech/substrate/issues/4437

Reply:
TY